tests/widget/widget.js

```diff
@@ -110,4 +112,85 @@ describe( 'Widget', () => {
 		sinon.assert.notCalled( focusSpy );
 		expect( getModelData( document ) ).to.equal( '[<widget></widget>]' );
 	} );
+
+	describe( 'delete and backspace handling', () => {
```

Comment:
> There are no tests for many potential false positive cases like what if the widget was selected in the first place and is preceded by another one, what if it's preceded by a paragraph with text or paragraph with an inline object, or what if we have inline object preceded by a text.

Deleting itself is handled by default delete mechanism I thought that it should be enough to check  only if modified behavior acts well here. But I think it is worth checking here too.